### PR TITLE
Prport of recent TBC changes

### PR DESCRIPTION
**Pull request by killerwife**
@namreeb

### CODE REVIEW

**Comment by killerwife** [comment]
Mistake found in https://github.com/cmangos/mangos-tbc/commit/530ece0af123c1670b80d46d0cc25933e1940aa4#commitcomment-20143498

Will be fixed today.

File: src/game/ObjectMgr.cpp [hunk 1]
```diff
@@ -3942,6 +3943,13 @@ void ObjectMgr::LoadQuests()
         }
         // else Skill quests can have 0 skill level, this is ok
 
+		if (qinfo->RequiredCondition)
```
Comment (Fabi) [inline src/game/ObjectMgr.cpp]: buuuh! tabs :D

File: src/game/Player.cpp [hunk 2]
```diff
@@ -13992,6 +13992,21 @@ bool Player::SatisfyQuestSkill(Quest const* qInfo, bool msg) const
...
+    if (uint32 conditionId = qInfo->GetRequiredCondition())
+    {
+        bool result = sObjectMgr.IsPlayerMeetToCondition(conditionId, this, GetMap(), nullptr, CONDITION_FROM_QUEST);
+
+        if(!result && msg)
+            SendCanTakeQuestResponse(INVALIDREASON_DONT_HAVE_REQ);
+
+        return result;
+    }
+    else
```
Comment (Fabi) [inline src/game/Player.cpp]: else needed here?

Comment (killerwife) [inline src/game/Player.cpp]: Just code design I guess. This way compiler knows for sure whats going on.

File: src/game/ObjectMgr.cpp (hunk 1 above)
Comment (killerwife) [inline src/game/ObjectMgr.cpp]: Will fix it, dunno what happened there. Thx.

**Comment by namreeb** [comment]
This got reverted (erased by force push) because it lacked an SQL revision update.

**Pull request merged**